Assets/UGSSamples/FriendsSample/Scripts/Common/Managers/RelationshipsManager.cs

```diff
@@ -148,9 +148,18 @@ async void AddFriendAsync(string name)
         {
             var success = await SendFriendRequest(name);
             if (success)
+            {
                 m_AddFriendView.FriendRequestSuccess();
+                if (m_RequestsEntryDatas.Find(entry => entry.Name == name) != null)

```

Comment:
I tested it, but I still had an error in the describe situation. If I understand correctly, the problems are the following:

1. add a friend
2. try to add it again (he did not accept the friend request yet)
3. Error
or
1. add a friend
2. he accepts
3. try add him again
4. error

Is it correct or are your describing another flow issue?

Reply:
This is tied to the flow above from line 228

If Player B has a pending request from Player A and they add them via the popup instead of clicking Accept, this check will see if a user is pending and then Refresh the UI accordingly.